src/CommandingOfficers/AWBW/OS/Andy.java

```diff
@@ -69,13 +71,15 @@ private static class HyperRepair extends AWBWAbility
     }
 
     @Override
-    protected void perform(MapMaster gameMap)
+    public GameEventQueue getEvents(MapMaster map)
     {
-      for( Unit u : myCommander.army.getUnits() )
-      {
-        u.alterHealthNoRound(HEAL*10);
-      }
-      super.perform(gameMap);
+      var heal = new MassHealEvent(null, myCommander.army.getUnits(), HEAL*10);
+      heal.roundUp = myCommander.roundUpRepairs;
+
+      GameEventQueue events = new GameEventQueue();
+      events.add(heal);
+
+      return events;
     }
   }
```

Comment:
Are we not using `perform` anymore? We may want to update some docstrings if we are changing how abilities are normally implemented.

Reply:
My thought was that events are the "preferred" mechanism, since they can be animated and listened to.
`perform` is still useful for stuff that doesn't have an associated event, though.

Reply:
Concur that `getEvent` should be preferred. Maybe `perform` will go away altogether eventually.